gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java

```diff
@@ -373,22 +495,23 @@ private WriteObjectRequest getCachedRequest(
     the GCS backend. The increase will be linear to number of chunks written, so that
     should also be fine.    */
     private void freeUpCommittedRequests(
-        TreeMap<Long, WriteObjectRequest> requestChunkMap, long writeOffset) throws IOException {
+        TreeMap<Long, WriteObjectRequest> requestChunkMap, long writeOffset, boolean block)
```

Comment:
Accepting boolean arguments for different behaviours is an anti-pattern. Please avoid it

Reply:
I don't follow. can you provide a reference to the anti-pattern.

Reply:
https://martinfowler.com/bliki/FlagArgument.html

Reply:
Thanks that was a good read. The anti pattern is more for public API. `freeupCommittedRequest` is a private function and if you see the implementation you will observe that it's hard to separate it out as two different methods without code duplication.